workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.ts

```diff
@@ -0,0 +1,12 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    toSet(this: Iterator<T>): Set<T>;
+  }
+}
+
+iteratorPrototype.toSet ??= function <T>(this: Iterator<T>): Set<T> {
```

Comment:
Btw, there is no `toSet` currently, and I don't think there's any plan for it at this time. So we can remove the `??` here and the `delete` from the test.

(I could potentially be convinced that we should keep the guards, so we're ready for a future when this method is added to the core language, but for now I'm inclined not to.)

Reply:
ok, sure!